Templates/DefaultComponent/template.json

```diff
@@ -23,13 +23,13 @@
             "isTemplated": true
         },
         {
-            "file": "Include/${Name}Interface.h",
+            "file": "Include/${GemName}/${Name}Interface.h",
```

Comment:
Well this makes the template DefaultComponent for a Gem template.
Now I think that 95% of new components would be made within gems, this should be OK.

If if were to make a component in a core library like AzFramework or AzToolsFramework, the "${GemName}" placeholder could be used.

Reply:
@lumberyard-employee-dm I went back and forth on whether or not to add this directory, and whether I should call the placeholder "GemName" or something more generic like "Namespace". The reason why I went with "GemName" is to provide the strong hint to the user of this template that this is normally the name of the Gem that they want to use here. I'm open to suggestions. I'm not sure by your comment, though, if you're suggesting that I not use the directory there, or maybe that I should not use the name "GemName" there.d Or are you OK with my doing both?

Reply:
Well the directory under the "Include" could be separated out from the name of the gem.

For non gem module libraries, a placeholder such as ${LibraryName} seems appropriate.
The drawback however of adding more placeholders is that it is another command line option that the user has to enter when instantiating the template.

This is fine for now, but I note this as a place that we may want to revisit if customers need more flexibility in configuring where a header files are located vs the namespace the source code uses.